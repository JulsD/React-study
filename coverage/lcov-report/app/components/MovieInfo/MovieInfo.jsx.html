<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/MovieInfo/MovieInfo.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/MovieInfo</a> MovieInfo.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { connect } from 'react-redux';
&nbsp;
export class MovieInfo extends React.Component {
&nbsp;
  constructor(props) {
    super(props);
  }
  
  render() {
    return (
      &lt;div&gt;
        Films by 
        {this.props.directors.map(item =&gt; &lt;span key={item.id}&gt;&lt;i&gt; &lt;/i&gt;{item.name}&lt;/span&gt;)}
      &lt;/div&gt;
    );
  }
&nbsp;
}
&nbsp;
function <span class="fstat-no" title="function not covered" >mapProps(</span>state) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    directors: state.movie.movie.director
  }
}
&nbsp;
export default connect(mapProps)(MovieInfo);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 09 2017 00:12:39 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
