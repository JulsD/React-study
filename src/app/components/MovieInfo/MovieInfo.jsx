import React from 'react';
import { connect } from 'react-redux';

class MovieInfo extends React.Component {

  constructor(props) {
    super(props);
  }
  
  render() {
    return (
      <div>
        Films by 
        {this.props.directors.map(item => <span key={item.id}><i> </i>{item.name}</span>)}
      </div>
    );
  }

}

function mapProps(state) {
  return {
    directors: state.movie.movie.director
  }
}

export default connect(mapProps)(MovieInfo);
