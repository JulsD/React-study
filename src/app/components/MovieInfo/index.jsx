import { default as MovieInfo } from './MovieInfo';
export default MovieInfo;