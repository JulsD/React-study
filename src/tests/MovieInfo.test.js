import React from 'react';
import { MovieInfo } from '../app/components/MovieInfo/MovieInfo';

test('MovieInfo should render a <div>', () => {
  const directors = [{id: 1, name: 'test'},{id:2, name: 'test'}, {id:3, name: 'test'}];
  const wrapper = shallow(
    <MovieInfo directors={directors}/>
  );
  expect(wrapper).toMatchSnapshot();
});
