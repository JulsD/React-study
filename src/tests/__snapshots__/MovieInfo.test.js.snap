// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo should render a <div> 1`] = `
<div>
  Films by
  <span
    key="1"
  >
    <i>
       
    </i>
    test
  </span>
  <span
    key="2"
  >
    <i>
       
    </i>
    test
  </span>
  <span
    key="3"
  >
    <i>
       
    </i>
    test
  </span>
</div>
`;
